app/assets/javascripts/TortoiseJS/agent/view.coffee

```diff
@@ -369,3 +283,251 @@ class PatchDrawer
     if world.patcheswithlabels
       @labelPatches(patches)
 
+
+class Line
+  constructor: (@x1, @y1, @x2, @y2) ->
+
+  midpoint: ->
+    midpointX = (@x1 + @x2) / 2
+    midpointY = (@y1 + @y2) / 2
+    [midpointX, midpointY]
+
+class LinkDrawer
+  constructor: (@view, @shapes) ->
+    directionIndicators = {}
+    for name, shape of @shapes
+      directionIndicators[name] = shape['direction-indicator']
+    @linkShapeDrawer = new ShapeDrawer(directionIndicators)
```

Comment:
My initial feeling is that this should inherit from `ShapeDrawer` instead, but I'm not sure there's a good enough reason to do so. Is there a reason why you didn't?